// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.status";
option java_outer_classname = "StatusProto";
option objc_class_prefix = "HLW";

package status;

// The greeting service definition.
service Status {
  // Sends status request
  rpc GetStatus (StatusReq) returns (StatusResp) {}
}

// The status request containing module name and optional data
message StatusReq {
  string module = 1;
  string optional_data = 2;
}

// The status request containing stdout, stderr and status code
message StatusResp {
  string stdout = 1;
  string stderr = 2;
  int32 status_code = 3;
}
